### PR TITLE
Jointly track x and y in PolygonSelector.

### PR DESCRIPTION
**Pull request by anntzer**
It's easier to track them in a single list.

Also init _selection_artist and _polygon_handles with empty arrays, as
there's no reason to pretend that they start with 0, 0.  On the other
hand, _xys does need to start as a non-empty array as the last point
gets updated as being the cursor position.

## PR Checklist

**Tests and Styling**
- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [ ] New features are documented, with examples if plot related.
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

**Comment by dstansby** [comment]
Can we wait until https://github.com/matplotlib/matplotlib/pull/21830 is merged for this to go in?

**Comment by anntzer** [comment]
Sure, sorry, I had only noticed #21881 as potentially conficting.

**Comment by ericpre** [comment]
#21830 has been merged, @anntzer, can you please rebase this PR?

**Comment by anntzer** [comment]
Thanks for the notification, rebased.

**Comment by QuLogic** [comment]
Seems to have broken some things.

**Comment by anntzer** [comment]
Thanks, I missed a point when rebasing over #21830.

**Review by dstansby — APPROVED** [review]
Much nicer, thanks a lot! This approval is contingent on CI passing.

File: lib/matplotlib/widgets.py [hunk 1]
```diff
@@ -3885,37 +3871,36 @@ def _on_key_release(self, event):
...
         else:
-            self._polygon_handles.set_data(self._xs, self._ys)
+            self._polygon_handles.set_data(xs, ys)
         self.update()
 
     @property
     def verts(self):
         """The polygon vertices, as a list of ``(x, y)`` pairs."""
-        return list(zip(self._xs[:-1], self._ys[:-1]))
+        return self._xys[:-1]
```
Comment (timhoffm) [inline lib/matplotlib/widgets.py:3903]: This now returns a handle to the internal data structure, which may have unintended side effects and could even break existing code. Should we return a copy?

Comment (anntzer) [reply lib/matplotlib/widgets.py:3903]: Actually `self._xys[:-1]` returns a copy of the `self._xys` list (it's not a ndarray).

**Review by timhoffm — APPROVED** [review]

File: lib/matplotlib/widgets.py (hunk 1 above)
Comment (timhoffm) [reply lib/matplotlib/widgets.py:3903]: Oh, yes of course.

**Pull request merged**